python/lsst/daf/butler/butler.py

```diff
@@ -173,9 +179,18 @@ def makeRepo(root, config=None, standalone=False, createRegistry=True, searchPat
         """
         if isinstance(config, (ButlerConfig, ConfigSubset)):
             raise ValueError("makeRepo must be passed a regular Config without defaults applied.")
-        root = os.path.abspath(root)
-        if not os.path.isdir(root):
-            safeMakeDir(root)
+
+        uri = ButlerURI(root)
+        if uri.scheme == "file":
```

Comment:
What about schemeless here?

Reply:
Following what you said (blindly)

> for "file" schemes we are assuming POSIX semantics for paths, for schemeless URIs we are assuming os.path semantics. 

I'd write the folowing:

```
...
uri = ButlerURI(root)
if uri.scheme is None:
    if not os.path.isdir(uri.ospath):
        safeMakeDir(uri.ospath)
if uri.scheme == "file":
    if not os.path.isdir(uri.path):
        safeMakeDir(uri.path)
...
```

but I'm confused as to what happens when someone uses a correctly specified URI on Widows. Then using simply `path` would not be correct for `file` schemes because its effectively a posix path and therefore needs to be `posix2os` converted before it would be usable again. So definitely for `file` schemes I need `ospath` and can't just assume posix style will work. If I am treating schemeless URIs as `os.path` and am on posix they will work as `file` schemes, and on windows they'll work again as `ospath`, then the two look exactly the same for all such `if`s? I'm not sure that's what you had imagined, but I went `if uri.scheme == "file" or not uri.scheme`

Reply:
`ButlerURI.ospath` was designed specifically for the case above so you could convert the logic to the final sentence in your comment above. That's why I suggested it. `file` URIs always use posix path separators (at least from what I can see reading URI RFCs).